pkg/operator/baremetal_pod.go

```diff
@@ -0,0 +1,397 @@
+package operator
+
+import (
+	"math/rand"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var baremetalConfigmap = "metal3-config"
+var volumeMounts = []corev1.VolumeMount{
+	{
+		Name:      "metal3-shared",
+		MountPath: "/shared",
+	},
+}
+
+func setEnvVar(name string, key string) corev1.EnvVar {
+	return corev1.EnvVar{
+		Name: name,
+		ValueFrom: &corev1.EnvVarSource{
+			ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: baremetalConfigmap,
+				},
+				Key: key,
+			},
+		},
+	}
+}
+
+var mariadbPassword = generateRandomPassword()
+
+func generateRandomPassword() string {
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789" +
+		"~!@#$%^&*()")
+	length := 8
+	buf := make([]rune, length)
+	for i := range buf {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return string(buf)
+}
+
+func setMariadbPassword() corev1.EnvVar {
+	return corev1.EnvVar{
+		Name: "MARIADB_PASSWORD",
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: "mariadb-password",
+				},
+				Key: mariadbPassword,
```

Comment:
This doesn't seem right ... it's now looking for a password from a random key name within the `mariadb-password` secret?

Reply:
Maybe you wanted to create a new Secret with a random password in it?  I think the code would be a bit different to do that

Reply:
IMO I think it'd be OK to just leave the secret as something that must be created outside of this patch and we address it as a follow-up

Reply:
For now, modified the code to read the password from an existing Secret that is created outside.